code/game/objects/items/devices/desynchronizer.dm

```diff
@@ -57,8 +57,9 @@
 	addtimer(CALLBACK(src, .proc/resync), duration)
 
 /obj/item/desynchronizer/proc/resync()
-	new /obj/effect/temp_visual/desynchronizer(sync_holder.drop_location())
-	QDEL_NULL(sync_holder)
+	if(sync_holder)
```

Comment:
Can you explain what's going on here

Reply:
sync_holder = null so sync_holder.drop_location runtimes